packages/docusaurus-theme-classic/src/theme/Navbar/index.js

```diff
@@ -45,8 +46,19 @@ function NavLink({
             to: toUrl,
             ...(activeBasePath
               ? {
-                  isActive: (_match, location) =>
-                    location.pathname.startsWith(activeBaseUrl),
+                  isActive: (_match, location) => {
+                    if (activeBasePathIsRegex) {
+                      return activeBasePath.test(location.pathname);
+                    }
+
+                    if (typeof activeBasePath === 'string') {
+                      return location.pathname.startsWith(activeBaseUrl);
+                    }
+
+                    throw new Error(
```

Comment:
We do not use throw errors on invalid type checking, I think this should be removed for consistency.

Reply:
Got it, will do